src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java

```diff
@@ -786,15 +790,38 @@ public void dynamicPVCVolume() throws Exception {
     private void dynamicPVC() throws Exception {
         assumePvcAccess();
         var client = cloud.connect();
-        var pods = client.pods().list().getItems();
-        var pvcs = client.persistentVolumeClaims().list().getItems();
+        SemaphoreStep.waitForStart("before/1", b);
+        var pods = getPodNames(client);
+        assertThat(pods, empty());
+        var pvcs = getPvcNames(client);
+        SemaphoreStep.success("before/1", null);
+        SemaphoreStep.waitForStart("pod/1", b);
+        assertThat(getPodNames(client), hasSize(1));
+        assertThat(getPvcNames(client), hasSize(1));
+        SemaphoreStep.success("pod/1", null);
         r.assertBuildStatusSuccess(r.waitForCompletion(b));
         await("The pods should be the same as before building")
                 .timeout(Duration.ofMinutes(1))
-                .until(() -> client.pods().list().getItems(), equalTo(pods));
+                .until(() -> getPodNames(client), equalTo(pods));
         await("The PVCs should be the same as before building")
                 .timeout(Duration.ofMinutes(1))
-                .until(() -> client.persistentVolumeClaims().list().getItems(), equalTo(pvcs));
+                .until(() -> getPvcNames(client), equalTo(pvcs));
```

Comment:
Looks like this might resolve https://github.com/jenkinsci/kubernetes-plugin/pull/1495#issuecomment-2110079900 (currently trying to reproduce locally).